ggml/src/ggml-cpu/unary-ops.cpp

```diff
@@ -95,25 +104,25 @@ static void apply_unary_op(const ggml_compute_params * params, ggml_tensor * dst
         dst_t        * dst_ptr  = (dst_t  *)       ((char *)       dst->data  + i03*nb3  + i02*nb2  + i01*nb1 );
         const src0_t * src0_ptr = (const src0_t *) ((const char *) src0->data + i03*nb03 + i02*nb02 + i01*nb01);
 
-        vec_unary_op<op>(ne0, dst_ptr, src0_ptr);
+        vec_unary_op<decltype(op), src0_t, dst_t>(op, ne0, dst_ptr, src0_ptr);
     }
 }
 
 // TODO: Use the 'traits' lookup table (for type conversion fns), instead of a mass of 'if' conditions with long templates
-template <float (*op)(float)>
-static void unary_op(const ggml_compute_params * params, ggml_tensor * dst) {
+template <typename Op>
+static void unary_op(const Op& op, const ggml_compute_params * params, ggml_tensor * dst) {
```

Comment:
Instead of passing the function pointer here (which comes with the risk of performance degrade), you can simply change the `op` signature from `float (*op)(float)` to `float (*op)(float, ggml_tensor * dst)`

Having both `template` and `decltype(op)` here is pretty much redundant as they will be all compiled into one single version of the function, which against the reason why they use them in the first place.